### PR TITLE
Add github template for pull requests

### PR DESCRIPTION
**Pull request by sauloperez**
This is a first proposal of template. It will show its contents when opening a PR, which will help providing the appropriate information for others to understand the changes.

You can read about it in https://github.com/blog/2111-issue-and-pull-request-templates

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#### What? Why?
+
+
+#### What should we test?
+
+
+#### How this relates to the Spree upgrade?
```
Comment (sauloperez) [inline .github/PULL_REQUEST_TEMPLATE.md]: How does...

**Comment by daniellemoorhead** [comment]
Hi @sauloperez @kirstenalarsen does this belong in a Spree milestone?

**Comment by sauloperez** [comment]
@daniellemoorhead we just need to discuss if we like it and merge it. This will only by github.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+#### How does this relate to the Spree upgrade?
+
+[remove me if not needed]
+
+#### Discourse thread
+
+[remove me if not needed]
+
+#### Dependencies
```
Comment (oeoeaio) [inline .github/PULL_REQUEST_TEMPLATE.md]: What sort of things are you anticipating will go in here?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#### What? Why?
+
+
+#### What should we test?
+
+
+#### How does this relate to the Spree upgrade?
```
Comment (oeoeaio) [inline .github/PULL_REQUEST_TEMPLATE.md]: Maybe something a bit more specific like: "Any known conflicts with the Spree Upgrade?" would be easier to describe? Just a thought, happy either way.

Comment (sauloperez) [inline .github/PULL_REQUEST_TEMPLATE.md]: Yep, I like being specific here.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (sauloperez) [inline .github/PULL_REQUEST_TEMPLATE.md]: When the PR depends on another one in the same or any other repo. The case I see (which might not happen often) is to depend on https://github.com/openfoodfoundation/ofn-install or a new feature requiring a refactor to be shipped first.

Yes, it needs to be described better.

**Comment by sauloperez** [comment]
I provided some instructions in each section. Hope this makes the template easier to fill.

**Pull request merged**